IPython/core/completer.py

```diff
@@ -1704,7 +1704,14 @@ def dispatch_custom_completer(self, text):
                  try_magic,
                  self.custom_completers.flat_matches(self.text_until_cursor)):
             try:
-                res = c(event)
+                try:
+                    res = c(event)
+                except KeyboardInterrupt:
```

Comment:
Does this need a nested try/except, or can we just add another except clause to the existing try/except?

Reply:
We can add an Except clause, but then it's hard to convey exactly where care about catching the keyboard interrupt. The only thing that could be slow is `c(event)`.But if you think that would be better I can move the clause.

Reply:
It feels neater to me to avoid another level of nesting, but I acknowledge there's also a benefit to your approach. I don't feel strongly about it, so let's leave it how you prefer it.